lib/DefaultFields.js

```diff
@@ -0,0 +1,226 @@
+ *
+ * @mixin
+ */
+class DefaultFields {
+    constructor() {
+        this.parent = null;
+        this.children = [];
+        this.fields = {};
+        this.parentFields = {};
+        this.precomputedFields = {};
+    }
+
+    /**
+     * This function allows to manually destroy a node, so that all references
+     * to parents or children are released. This helps in avoiding the creation
+     * of memory leaks due to our internal cyclic references that cannot be
+     * automatically released due to the mechanics of GC in Node.JS, and its
+     * shitty prevention of meaningful OOP designs.
+     *
+     * All hail << The Coming of the Great White Handkerchief >> that will
+     * wipe this misery out of our lives !
+     *
+     * @returns {undefined}
+     */
+    _dfDestroy() {
+        this.children.forEach(child => {
+            child.setParent(null);
+        });
+        this.setParent(null);
+    }
+
+    /**
+     * Add a node to the list of children of the current node, effectively
+     * making it one of the node's children.
+     *
+     * @param {DefaultFields} child - The node to add as a child of the current
+     *                                node
+     *
+     * @return {undefined}
+     */
+    _dfRegisterChild(child) {
+        this.children.push(child);
+        this._dfNotifyChild(child);
+    }
+
+    /**
+     * Remove a node from the list of children of the current node, effectively
+     * cutting off the relationship between the two.
+     *
+     * @param {DefaultFields} toRemove - The node to remove from the list of
+     *                                   children of the current node.
+     *
+     * @return {undefined}
+     */
+    _dfUnregisterChild(toRemove) {
+        const filtered = this.children.filter(child => child !== toRemove);
+        this.children = filtered;
+    }
+
+    /**
+     * Utility function to notify one child node of an update of the node's
+     * precomputed fields.
+     *
+     * @param {DefaultField} child - The child node to notify
+     *
+     * @return {undefined}
+     */
+    _dfNotifyChild(child) {
+        child._dfSetParentFields(this.precomputedFields);
+    }
+
+    /**
+     * Utility function to notify every children node of an update of the
+     * node's precomputed fields.
+     *
+     * @return {undefined}
+     */
+    _dfNotifyChildren() {
+        this.children.forEach(child => this._dfNotifyChild(child));
+    }
+
+    /**
+     * This function allows changing the Parent node of the current node,
+     * consequently changing the resulting aggregation of the hierarchy of
+     * fields. This can be used for a temporary switch of parent node.
+     *
+     * @param {DefaultFields} parent - The new parent node to set for the
+     *                                 current node
+     *
+     * @return {DefaultFields|null} The previous parent
+     */
+    setParent(parent) {
+        const oldParent = this.parent;
```

Comment:
Indeed, as the `_dfRemoveChild()` function was not intended to be called by anything other than internal mechanisms. As such, what you suggests seems to be a protection of sorts against circular dependency between the two functions (each calling the other) to avoid unintended  infinite recursion.
If you think it's useful to do this, I can, although I didn't see much of a point when writing this code.